File: android/app/src/main/java/com/alexivanov/snapsell/domain/Money.kt
package com.alexivanov.snapsell.domain

import java.util.Locale

/** USD display helpers shared by the UI and the clipboard block. */
object Money {
    /** "75" for whole dollars, "79.99" otherwise. No currency symbol. */
    fun plain(amount: Double): String =
        if (amount == Math.floor(amount) && !amount.isInfinite()) {
            String.format(Locale.US, "%d", amount.toLong())
        } else {
            String.format(Locale.US, "%.2f", amount)
        }

    /** "$75" / "$79.99". */
    fun usd(amount: Double): String = "$" + plain(amount)
}

File: ios/SnapSell/Domain/Money.swift
import Foundation

/// USD display helpers shared by the UI and the clipboard block.
enum Money {
    /// "75" for whole dollars, "79.99" otherwise. No currency symbol.
    static func plain(_ amount: Double) -> String {
        if amount == amount.rounded(.down), amount.isFinite {
            return String(Int64(amount))
        }
        return String(format: "%.2f", locale: Locale(identifier: "en_US_POSIX"), amount)
    }

    /// "$75" / "$79.99".
    static func usd(_ amount: Double) -> String { "$" + plain(amount) }
}
